h/static/scripts/search-client.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+var EventEmitter = require('tiny-emitter');
+var inherits = require('inherits');
+
+/**
+ * Client for the Hypothesis search API.
+ *
+ * SearchClient handles paging through results, canceling search etc.
+ *
+ * @param {Object} resource - ngResource class instance for the /search API
```

Comment:
As a separate task, I think we can look at replacing `ngResource` with a simple fetch here. I'll have to check exactly what it does for us at the moment that we still rely on.